libcalico-go/lib/apis/v3/ipam_block.go

```diff
@@ -39,15 +39,46 @@ type IPAMBlock struct {
 
 // IPAMBlockSpec contains the specification for an IPAMBlock resource.
 type IPAMBlockSpec struct {
-	CIDR           string                `json:"cidr"`
-	Affinity       *string               `json:"affinity,omitempty"`
-	StrictAffinity bool                  `json:"strictAffinity"`
-	Allocations    []*int                `json:"allocations"`
-	Unallocated    []int                 `json:"unallocated"`
-	Attributes     []AllocationAttribute `json:"attributes"`
+	// The block's CIDR.
+	CIDR string `json:"cidr"`
 
+	// Affinity of the block, if this block has one. If set, it will be of the form
+	// "host:<hostname>". If not set, this block it not affine to a host.
+	Affinity *string `json:"affinity,omitempty"`
+
+	// Array of allocations in-use within this block. nil entries mean the allocation is free.
+	// For non-nil entries at index i, the index is the ordinal of the allocation within this block
+	// and the value is the index of the associated attributes in the Attributes array.
+	Allocations []*int `json:"allocations"`
+
+	// Unallocated is an ordered list of allocations which are free in the block.
+	Unallocated []int `json:"unallocated"`
+
+	// Attributes is an array of arbitrary metadata associated with allocations in the block. To find
+	// attributes for a given allocation, use the value of the allocation's entry in the Allocations array
+	// as the index of the element in this array.
+	Attributes []AllocationAttribute `json:"attributes"`
+
+	// We store a sequence number that is updated each time the block is written.
```

Comment:
Something to consider is if we just want to use the Revision for this rather than a new field. 

I think this would work OK, so long as we don't create a block with an allocation already in it (i.e., allocations must be done with updates not creates). The problem being that the Revision doesn't exist until the block is created, and we mustn't initialize allocations with the same sequence number otherwise you can hit a corner case where allocating / releasing an IP results in the same sequence number (if the block is deleted inbetween)